vsintegration/src/FSharp.LanguageService/XmlDocumentation.fs

```diff
@@ -23,49 +24,32 @@ type internal TextSanitizingCollector(collector, ?lineLimit: int) =
     let mutable endsWithLineBreak = false
     let mutable count = 0
 
-    let buf = StringBuilder()
-
-    let addTaggedTextEntry t =
-        if lineLimit.IsNone || count < lineLimit.Value then 
-            isEmpty <- false
-            endsWithLineBreak <- match t with TaggedText.LineBreak _ -> true | _ -> false
-            if endsWithLineBreak then count <- count + 1
-            collector t
-        if lineLimit.IsSome && lineLimit.Value = count then
+    let addTaggedTextEntry text =
+        match lineLimit with
+        | Some lineLimit when lineLimit = count ->
             // add ... when line limit is reached
             collector (tagText "...")
             count <- count + 1
+        | _ ->
+            isEmpty <- false
+            endsWithLineBreak <- match text with TaggedText.LineBreak _ -> true | _ -> false
+            if endsWithLineBreak then count <- count + 1
+            collector text
     
-    let isCROrLF c = c = '\r' || c = '\n'
+    let splitTextRegex = Regex(@"\s*\n\s*\n\s*")
```

Comment:
It's used to split 1-10 lines at a time, triggered by mouse move... It's not a place where we should think of any optimizations.